File: VB/Sample/Models/Person.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel.DataAnnotations

Namespace Sample.Models
	Public Class Person
		Public Sub New()
			PersonID = 0
			FirstName = String.Empty
			MiddleName = String.Empty
			LastName = String.Empty
		End Sub

		Public Sub New(ByVal id As Integer, ByVal firstName As String, ByVal middleName As String, ByVal lastName As String)
			Me.PersonID = id
			Me.FirstName = firstName
			Me.MiddleName = middleName
			Me.LastName = lastName
		End Sub

		Private privatePersonID As Integer
		<Key> _
		Public Property PersonID() As Integer
			Get
				Return privatePersonID
			End Get
			Set(ByVal value As Integer)
				privatePersonID = value
			End Set
		End Property

		Private privateFirstName As String
		<Required(ErrorMessage := "First Name is required")> _
		Public Property FirstName() As String
			Get
				Return privateFirstName
			End Get
			Set(ByVal value As String)
				privateFirstName = value
			End Set
		End Property

		Private privateMiddleName As String
		Public Property MiddleName() As String
			Get
				Return privateMiddleName
			End Get
			Set(ByVal value As String)
				privateMiddleName = value
			End Set
		End Property

		Private privateLastName As String
		<Required(ErrorMessage := "Last Name is required")> _
		Public Property LastName() As String
			Get
				Return privateLastName
			End Get
			Set(ByVal value As String)
				privateLastName = value
			End Set
		End Property
	End Class
End Namespace

File: CS/Sample/Models/Person.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Sample.Models {
    public class Person {
        public Person() {
            PersonID = 0;
            FirstName = string.Empty;
            MiddleName = string.Empty;
            LastName = string.Empty;
        }

        public Person(int id, String firstName, string middleName, String lastName) {
            this.PersonID = id;
            this.FirstName = firstName;
            this.MiddleName = middleName;
            this.LastName = lastName;
        }

        [Key]
        public int PersonID { get; set; }

        [Required(ErrorMessage = "First Name is required")]
        public string FirstName { get; set; }

        public string MiddleName { get; set; }

        [Required(ErrorMessage = "Last Name is required")]
        public string LastName { get; set; }
    }
}